### PR TITLE
consensus: realize BIP 100

### PR DESCRIPTION
**Pull request by andyoknen**
## Standards checklist:

- [X] The PR title is descriptive.
- [X] The PR doesn't replicate another PR which is already open.
- [X] The PR has self-explained commits history.

- [X] The code is mine or it's from somewhere with an Apache-2.0 compatible license.
- [X] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [X] If the code introduces new classes, methods, I provide a valid use case for all of them.

## Changes:

[BIP 100](https://github.com/pocketnetteam/pocketnet.core/wiki/BIP-100:-Changes-in-emission-payments)

### CODE REVIEW

File: src/pocketdb/consensus/Lottery.cpp [hunk 1]
```diff
@@ -202,6 +202,58 @@ namespace PocketConsensus
...
+        {
+            // Get destination address and score value
+            // In lottery allowed only likes to posts and comments
+            // Also in lottery allowed only positive scores
+            auto[parseScoreOk, scoreTxData] = TransactionHelper::ParseScore(tx);
+            if (!parseScoreOk)
+                continue;
+
+            // BIP100: Only scores to content allowed
+            if (scoreTxData->ScoreType == ACTION_SCORE_CONTENT
```
Comment (lostystyg) [inline src/pocketdb/consensus/Lottery.cpp:233]: Shouldn't it be `scoreTxData->ScoreType != ACTION_SCORE_CONTENT`?

Comment (lostystyg) [reply src/pocketdb/consensus/Lottery.cpp:233]: Or even `scoreTxData->ScoreType != ACTION_SCORE_CONTENT || (scoreTxData->ScoreValue != 4 && scoreTxData->ScoreValue != 5)` if i understand the logic correctly.

Comment (andyoknen) [reply src/pocketdb/consensus/Lottery.cpp:233]: @lostystyg Thank you! This is a valuable note, I added a correction (for clarity, I added an additional condition to exclude comments in this place)

Comment (andyoknen) [reply src/pocketdb/consensus/Lottery.cpp:233]: I checked on the test network - the last change did not cause compatibility problems. I'm closing the question

**Review by only1question — APPROVED** [review]

**Review by lostystyg — APPROVED** [review]

**Review by zedxter — APPROVED** [review]

**Pull request merged**